app/api/discount_codes.py

```diff
@@ -3,47 +3,79 @@
 from sqlalchemy.orm.exc import NoResultFound
 
 from app.api.helpers.db import safe_query
-from app.api.helpers.exceptions import UnprocessableEntity, ForbiddenException
+from app.api.helpers.exceptions import (
+    UnprocessableEntity, ForbiddenException, ConflictException)
 from app.api.helpers.permission_manager import has_access
 from app.api.helpers.permissions import jwt_required, current_identity
 from app.api.helpers.utilities import require_relationship
-from app.api.schema.discount_codes import DiscountCodeSchemaTicket, DiscountCodeSchemaEvent
+from app.api.schema.discount_codes import DiscountCodeSchemaTicket, DiscountCodeSchemaEvent, DiscountCodeSchemaPublic
 from app.models import db
 from app.models.discount_code import DiscountCode
 from app.models.event import Event
 from app.models.event_invoice import EventInvoice
 from app.models.user import User
-
+from app.models.ticket import Ticket
 
 class DiscountCodeListPost(ResourceList):
     """
     Create Event and Ticket Discount code and Get Event Discount Codes
     """
 
+    def decide_schema(self, json_data):
+        """To decide discount code schema based on posted data.
+        :param json_data:
+        :return:"""
+
+        used_for = json_data['data']['attributes'].get('used-for')
+        if used_for in ('event', 'ticket'):
+            if used_for == 'event':
+                self.schema = DiscountCodeSchemaEvent
+            elif used_for == 'ticket':
+                self.schema = DiscountCodeSchemaTicket
+        else:
+            raise ConflictException(
+                {'pointer': '/data/attributes/used-for'},
+                "used-for attribute is required to create discount code")
```

Comment:
If `used-for` is equal to "random", then your exception message will say that `used-for attribute is required to create discount code`.